src/room-object.ts

```diff
@@ -28,13 +28,39 @@ interface RoomObjectConstructor extends _Constructor<RoomObject> {
 
 declare const RoomObject: RoomObjectConstructor;
 
-interface RoomObjectEffect {
+/**
+ * Discriminated union of possible effects on `effect`
+ */
+type RoomObjectEffect = NaturalEffect | PowerEffect;
+
+/**
+ * Natural effect applied to room object
+ */
+interface NaturalEffect {
+    /**
+     * Effect ID of the applied effect. `EFFECT_*` constant.
+     */
+    effect: EffectConstant;
+    /**
+     * How many ticks will the effect last.
+     */
+    ticksRemaining: number;
+}
+
+/**
+ * Effect applied to room object as result of a `PowerCreep.usePower`.
+ */
+interface PowerEffect {
     /**
      * Power level of the applied effect.
      */
     level: number;
     /**
-     * Power ID of the applied effect. `PWR_*` constant.
+     * Effect ID of the applied effect. `PWR_*` constant.
+     */
+    effect: PowerConstant;
+    /**
+     * @deprecated Power ID of the applied effect. `PWR_*` constant. No longer documented, will be removed.
```

Comment:
I have checked in the doc and in-game, and it seems to be the contrary: 
- In PowerEffect, "effect" should be deprecated
- In NaturalEffect, "power" should be deprecated
Or am I missing something?

Reply:
@pyrodogg Yes you are right, but I am confused because in the documentation, if you go see a specific RoomObject, it says `Either power or effect property is present`... (https://docs.screeps.com/api/#StructureController.effects)
Anyway, this is ok as it is, and we could still refine it later on.